### ISSUE TITLE
WifiForIotPlugin.loadWifiList() always returns empty list. in Android 10

### ISSUE THREAD

**Issue by kausik-dev** [flutternetwork/WiFiFlutter#184]

**Comment by muhammedkamal** [comment]
same problem

**Comment by muhammedkamal** [comment]
any updates?

**Comment by fernando-s97** [comment]
Same problem here. Maybe it could be related to #129?

**Comment by daadu** [comment]
1. Make sure that location permission is granted by user.
2. Use `WiFiForIoTPlugin.onWifiScanResultReady` with `WiFiForIoTPlugin.loadWifiList`

This is how I do it in my proj:


```dart
class _WifiSSIDPickDialogState extends State<WifiSSIDPickDialog> {
  bool isScanning = true;
  List<WifiNetwork> networks;

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) => scan(context));
  }

  Future<void> scan(BuildContext context) async {
    if (await _checkAndRequestLocationPermission()) {
      setState(() => isScanning = true);
      final scannedNetwork = await WiFiForIoTPlugin.loadWifiList();
      setState(() => networks = scannedNetwork);
      setState(() => isScanning = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return StreamBuilder<List<WifiNetwork>>(
      stream: WiFiForIoTPlugin.onWifiScanResultReady,
      builder: (context, snapshot) {
        final _networks = snapshot?.data ?? networks ?? [];
      },
    );
  }
}
```

Unofruntaley, this is not documented anywhere. If it works for you then any PR for doc improvement is welcomed.

**Comment by fernando-s97** [comment]
@daadu I didn't know the location permission, but I didn't understand the relation between the `loadWifiList` and the `onWifiScanResultReady`. Do I need to call `loadWifiList` and get the response in `onWifiScanResultReady`?

**Comment by daadu** [comment]
> @daadu I didn't know the location permission, but I didn't understand the relation between the `loadWifiList` and the `onWifiScanResultReady`. Do I need to call `loadWifiList` and get the response in `onWifiScanResultReady`?

Yes, that is what is working for me.

**Comment by kausik-dev** [comment]
@daadu  I've tried your solution , but still the list  is  empty.

**Comment by daadu** [comment]
Note: Location needs to be enabled for the scan to happen.

**Comment by kausik-dev** [comment]
> Note: Location needs to be enabled for the scan to happen.

Even after enabling the location , the same problem occurs.

**Comment by daadu** [comment]
re-written new plugin - currently not released - PR #205 - encourage you to give review, test and give feedback on it.

This issue will be linked to that plugin and will be closed with this PR.
